src/rate/coincap.js

```diff
@@ -15,30 +31,41 @@ export function makeCoincapPlugin(opts: EdgeCorePluginOptions): EdgeRatePlugin {
     },
 
     async fetchRates(pairsHint) {
-      const reply = await io.fetch('https://api.coincap.io/v2/assets?limit=500')
-      const json = await reply.json()
-
-      // Grab all the pairs which are in USD:
       const pairs = []
-      for (const entry of json.data) {
-        if (typeof entry.symbol !== 'string') continue
-        if (typeof entry.priceUsd !== 'string') continue
-        const currency = entry.symbol
+      if (Object.keys(currencyMap).length === 0) {
+        const assets = await io.fetch(`https://api.coincap.io/v2/assets/`)
+        const assetsJson = await assets.json()
+        const assetIds = asCoincapAssets(assetsJson.data)
+        assetIds.forEach(code => (currencyMap[code.symbol] = code.id))
+        log(`rate data coincap ${JSON.stringify(currencyMap)}`)
+      }
+      for (const pair of pairsHint) {
+        // Coincap only provides prices in USD and must be queried by unique identifier rather that currency code
+        if (!currencyMap[pair.fromCurrency] || pair.toCurrency !== 'iso:USD')
```

Comment:
It is still worth returning The pair even if it's not USD. The core can find A non USD rate by routing through A fiat exchange